core/constant_inc.php

```diff
@@ -591,8 +592,11 @@
 define( 'DB_FIELD_SIZE_PASSWORD', 64 );
 define( 'DB_FIELD_SIZE_API_TOKEN_NAME', 128 );
 
-# Maximum size for the user's password when storing it as a hash
-define( 'PASSWORD_MAX_SIZE_BEFORE_HASH', 1024 );
+# Maximum size for the user's password when storing it as a hash.
+# The CRYPT_BLOWFISH algorithm used by password_hash() will silently truncate
+# passwords to 72 characters, so there is no point in allowing longer passwords.
+# @see http://php.net/function.crypt
+define( 'PASSWORD_MAX_SIZE_BEFORE_HASH', 72 );
```

Comment:
> Per our release management guidelines, removing deprecated feature should only happen in Major versions.

I mean eventually we should retire plain password when it is the right time, till then a good admin check is a good option to use.

> IIRC we already do that on the front-end (via input field maxlength).

I recall that we had a fixed max length.  The max length in case of using a hash is not really useful.  But if we are using plain passwords, it is must to apply a 64 limit.  Hence, we should have an app for max length to enforce in the UI that should take into consideration the algorithm used.